app/controllers/admin/username_blocks_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class Admin::UsernameBlocksController < Admin::BaseController
```

Comment:
The entire admin area workflow of viewing, creating, editing, etc - could probably use a new admin system spec?